helios-retrofit-integration/src/main/kotlin/helios/retrofit/HeliosConverterFactory.kt

```diff
@@ -0,0 +1,49 @@
+package helios.retrofit
+
+import arrow.core.Option
+import arrow.core.Tuple2
+import arrow.core.getOrElse
+import arrow.data.extensions.list.monad.map
+import helios.typeclasses.Decoder
+import helios.typeclasses.Encoder
+import okhttp3.RequestBody
+import okhttp3.ResponseBody
+import retrofit2.Converter
+import retrofit2.Retrofit
+import java.lang.reflect.Type
+
+class HeliosConverterFactory private constructor(
+  private val decoderInstances: Map<String, Decoder<*>>,
+  private val encoderInstances: Map<String, Encoder<*>>
+) : Converter.Factory() {
+
+  companion object {
+    fun create(jsonables: List<Triple<Class<*>, Encoder<*>, Decoder<*>>>): HeliosConverterFactory {
```

Comment:
I would make a simple abstraction that at least enforces that the triple are all for the same type. Not _all triples_ in the list, just each triple among itself, and the list accepts `*` versions of it.

Reply:
Would something like `typealias JsonableEvidence<T> = Triple<Class<T>, Encoder<T>, Decoder<T>>` be ok?

Reply:
Yep!